src/crimson/osd/pg_recovery.cc

```diff
@@ -605,8 +614,12 @@ void PGRecovery::update_peers_last_backfill(
 
 bool PGRecovery::budget_available() const
 {
-  // TODO: the limits!
-  return true;
+  auto max = crimson::common::get_conf<
```

Comment:
Continuing my previous, I think we should be able to reuse `osd_recovery_max_single_start` here.
Let's also add a log line here with the three counters/limits to be aware that throttling has failed.

"max_active_push_per_reactor {}, max_active_push_per_osd {}, ongoing_pushes {}"